common/dal2/tests/it/services/s3.rs

```diff
@@ -26,6 +27,34 @@ async fn builder() {
         .credential(Credential::hmac("access-key", "secret-key"))
         .endpoint("http://localhost:9000")
         .finish()
+        .unwrap();
+}
+
+#[tokio::test]
+// TODO: we need to add an integration test for it.
+#[ignore]
+async fn test_read() {
+    let mut builder = s3::Backend::build();
+
+    let da = builder
+        .root("tests")
+        .bucket("testbucket")
+        .region("us-east-1")
+        .credential(Credential::hmac("minioadmin", "minioadmin"))
+        .endpoint("http://localhost:9900")
+        .finish()
+        .unwrap();
+
+    let op = Operator::new(da);
+
+    let mut r = op
+        .read("data/sample.csv")
+        .size(4 * 1024 * 1024)
+        .run()
         .await
         .unwrap();
+
+    let mut buf: Vec<u8> = Vec::new();
+    r.read_to_end(&mut buf).await.unwrap();
+    println!("{}", String::from_utf8(buf).unwrap());
```

Comment:
```suggestion

```

Reply:
Those tests will be refactor into a behavior test for all DAL services.